### DISCUSSION TITLE
SyntaxError: JSON Parse error: Unexpected EOF

### DISCUSSION THREAD

**Discussion by fusionconsulting** [livewire/livewire#7864] [1. Help] [answered]
I have a Livewire component for listing database entries of various types. I have used the same component for 12 types and all work fine, however one component does not work and in the console I have the error:

SyntaxError: JSON Parse error: Unexpected EOF

![Screenshot 2024-01-31 at 17 11 17](https://github.com/livewire/livewire/assets/3878232/61d0a9e0-0888-45c2-bfd6-6d22cf70c451)

The line in the file is:

this.snapshotEncoded = el.getAttribute("wire:snapshot");
this.snapshot = JSON.parse(this.snapshotEncoded); <-- this is the source of the error

If I put a breakpoint on the offending line, I can see that in this case "snapshotEncoded" is just an empty string but contains data for all other components. I have reduced the component view to simply <div></div> and still see the same error but only on this one component.

So, my question is firstly, is it valid to have an empty string as the snapshotEncoded value? If so, then I will continue digging. If not, and this is the cause of the problem, what would cause it to be empty?

I have tried to create a simple project to show the issue but it doesn't happen outside of this one single component so I'm stumped at this point as to what could cause it.

**Comment by fusionconsulting** [comment, marked as answer]
So this turned out to be completely my fault, staring at it for so long I couldn't figure it out. Back to basics and discovered a bug in a Wireable model.

**Comment by sajanp** [comment]
It's insane to me when people do this.

"Found the issue, was my fault."

Frustrating future developers who arrive here by search for years and decades to come.  DenverCoder19, why must you behave this way.
